--- a/pexpect.py	2014-05-07 22:02:46.000000000 +0000
+++ b/pexpect.py	2014-05-08 02:55:07.018969340 +0000
@@ -1134,7 +1134,7 @@
     """
     # Special case where filename already contains a path.
     if os.path.dirname(filename) != '':
-        if os.access (filename, os.X_OK) and not os.path.isdir(f):
+        if os.access (filename, os.X_OK) and not os.path.isdir(filename):
             return filename
 
     if not os.environ.has_key('PATH') or os.environ['PATH'] == '':
