diff -rup libgap-p/src/src/Makefile.in libgap-4.6.4.p0/src/src/Makefile.in
--- src/src/Makefile.in	2013-07-13 22:18:07.000000000 +0200
+++ src/src/Makefile.in	2013-06-04 15:10:00.000000000 +0200
@@ -300,7 +300,7 @@ libgap_la_SOURCES = ariths.c c_random.c
 	compstat.c gap.c listoper.c plist.c streams.c c_oper1.c \
 	lists.c precord.c string.c costab.c gasman.c macfloat.c \
 	range.c sysfiles.c $(am__append_1)
-libgapincludedir = @includedir@/gap
+libgapincludedir = $(pkgincludedir)
 libgapinclude_HEADERS = \
   ariths.h    cyclotom.h  integer.h   objcftl.h   rational.h  sysfiles.h \
   blister.h   dteval.h    intfuncs.h  objects.h   read.h      system.h   \
