diff --git a/src/language/init.c b/src/language/init.c
index 8473a3bfc..54053eaaf 100644
--- a/src/language/init.c
+++ b/src/language/init.c
@@ -854,6 +854,11 @@ static void *
 pari_mainstack_malloc(size_t size)
 {
   void *b;
+
+  /* Sage-specific patch to disable the following check on Cygwin--this check
+   * causes significant slowdown on Cygwin for not much benefit, so we disable
+   * it; see Trac #28132 */
+#ifndef __CYGWIN__
   /* Check that the system allows reserving "size" bytes. This is just
    * a check, we immediately free the memory. */
   BLOCK_SIGINT_START;
@@ -862,6 +867,9 @@ pari_mainstack_malloc(size_t size)
   if (b == MAP_FAILED) return NULL;
   BLOCK_SIGINT_START;
   munmap(b, size);
+#else
+  BLOCK_SIGINT_START;
+#endif
 
   /* Map again, this time with MAP_NORESERVE. On some operating systems
    * like Cygwin, this is needed because remapping with PROT_NONE and
@@ -884,8 +892,8 @@ pari_mainstack_mfree(void *s, size_t size)
 /* Completely discard the memory mapped between the addresses "from"
  * and "to" (which must be page-aligned).
  *
- * We use mmap() with PROT_NONE, which means that the underlying memory
- * is freed and that the kernel should not commit memory for it. We
+ * We (generally) use mmap() with PROT_NONE, which means that the underlying
+ * memory is freed and that the kernel should not commit memory for it. We
  * still keep the mapping such that we can change the flags to
  * PROT_READ|PROT_WRITE later.
  *
@@ -893,7 +901,12 @@ pari_mainstack_mfree(void *s, size_t size)
  * calling mprotect(..., PROT_NONE) because the latter will keep the
  * memory committed (this is in particular relevant on Linux with
  * vm.overcommit = 2). This remains true even when calling
- * madvise(..., MADV_DONTNEED). */
+ * madvise(..., MADV_DONTNEED).
+ *
+ * Unfortunately, the opposite is true on Cygwin--there, calling
+ * mprotect(..., PROT_NONE) does in fact decommit / free the mapped
+ * memory. Meanwhile there is a known bug that prevents already mapped
+ * memory to be re-mapped with MAP_FIXED and different protection flags. */
 static void
 pari_mainstack_mreset(pari_sp from, pari_sp to)
 {
@@ -902,11 +915,17 @@ pari_mainstack_mreset(pari_sp from, pari_sp to)
   if (!s) return;
 
   addr = (void*)from;
+#ifdef __CYGWIN__
+  BLOCK_SIGINT_START;
+  if (mprotect(addr, s, PROT_NONE)) pari_err(e_MEM);
+  BLOCK_SIGINT_END;
+#else
   BLOCK_SIGINT_START;
   res = mmap(addr, s, PROT_NONE,
              MAP_FIXED|MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0);
   BLOCK_SIGINT_END;
   if (res != addr) pari_err(e_MEM);
+#endif
 }
 
 /* Commit (make available) the virtual memory mapped between the
