diff --git a/config/extract_files b/config/extract_files
index e8eae1a..2119dee 100755
--- a/config/extract_files
+++ b/config/extract_files
@@ -9,7 +9,7 @@ for dir in "$doc_dir" "$misc_dir"; do
     echo "...in $dir"
     flist=`ls $dir/*.in`
     for file in $flist; do
-      sed -e "s%@perl@%$perl%g"\
+      sed -e "s%@perl@%/usr/bin/env perl%g"\
           -e "s%@datadir@%$datadir%g"\
           -e "s%@bindir@%$bindir%g"\
           -e "s%@includedir@%$includedir%g"\
