diff -Naur numpy.orig/distutils/system_info.py numpy/distutils/system_info.py
--- a/numpy/distutils/system_info.py	2016-03-10 22:30:23.003459873 +1300
+++ b/numpy/distutils/system_info.py	2016-03-11 14:17:30.255557025 +1300
@@ -589,6 +589,7 @@
     def get_paths(self, section, key):
         dirs = self.cp.get(section, key).split(os.pathsep)
         env_var = self.dir_env_var
+        print("from get_paths %s - %s"%(dirs,env_var))
         if env_var:
             if is_sequence(env_var):
                 e0 = env_var[-1]
@@ -628,9 +629,11 @@
                                 ds2.append(d1)
                 dirs = ds2 + dirs
         default_dirs = self.cp.get(self.section, key).split(os.pathsep)
+        print("from get_paths, default_dirs %s"%default_dirs)
         dirs.extend(default_dirs)
         ret = []
         for d in dirs:
+	    print("from get_paths, dir loop- d= %s"%d)
             if not os.path.isdir(d):
                 warnings.warn('Specified path %s is invalid.' % d)
                 continue
@@ -695,6 +698,7 @@
         """
         exts = self.library_extensions()
         info = None
+        print("from check_libs: %s libs %s"%(lib_dirs,libs))
         for ext in exts:
             info = self._check_libs(lib_dirs, libs, opt_libs, [ext])
             if info is not None:
@@ -749,6 +753,7 @@
         """
         # First, try to find the mandatory libraries
         if is_sequence(lib_dirs):
+	    print("is sequence %s"%lib_dirs)
             found_libs, found_dirs = [], []
             for dir_ in lib_dirs:
                 found_libs1 = self._lib_list(dir_, libs, exts)
@@ -763,6 +768,7 @@
                         if dir_ not in found_dirs:
                             found_dirs.append(dir_)
         else:
+	    print("not sequence %s"%lib_dirs)
             found_libs = self._lib_list(lib_dirs, libs, exts)
             found_dirs = [lib_dirs]
         if len(found_libs) > 0 and len(found_libs) == len(libs):
