eclib patch from https://github.com/JohnCremona/eclib/pull/29

diff --git a/libsrc/eclib/interface.h b/libsrc/eclib/interface.h
index b8eff26..127b48e 100644
--- a/libsrc/eclib/interface.h
+++ b/libsrc/eclib/interface.h
@@ -161,8 +161,7 @@ inline int is_approx_zero(const RR& x)
 }
 } // namespace NTL
 #ifdef _LIBCPP_VERSION
-namespace std {
-inline namespace __1 {
+namespace NTL {
 inline bool isinf(const RR& z) {
   return false;
 }
@@ -183,7 +182,6 @@ inline RR fmax(const RR& x, const RR& y)
   return x < y ? y : x;
 }
 }
-}
 #endif
 
 #include <complex>
 