--- a/pyproject.toml	2021-12-01 10:23:30.000000000 -0300
+++ b/pyproject.toml	2021-12-13 18:01:36.239921204 -0300
@@ -3,7 +3,6 @@
 requires=[
   "setuptools",
   "wheel",
-  "debugpy",
   "ipython>=5",
   "jupyter_core>=4.2",
   "jupyter_client",
diff -ruN a/setup.py b/setup.py
--- a/setup.py	2021-12-01 10:23:09.000000000 -0300
+++ b/setup.py	2021-12-13 18:01:40.112873823 -0300
@@ -70,5 +70,4 @@
     python_requires=">=3.7",
     install_requires=[
-        "debugpy>=1.0",
         "ipython>=7.23.1",
         "traitlets>=5.1.0",
