--- a/pGroupCohomology-3.3.2/setup.py	2021-11-04 10:32:09.000000000 -0500
+++ b/pGroupCohomology-3.3.2/setup.py	2021-11-04 10:32:14.000000000 -0500
@@ -38,26 +38,27 @@
 
 # Sage >= 6.8
 from sage.env import sage_include_directories
+includes = sage_include_directories() + ['.']
 
 ext_mods = [
     Extension("pGroupCohomology.resolution",
               sources = [os.path.join("pGroupCohomology","resolution.pyx")],
-              include_dirs = sage_include_directories(),
+              include_dirs = includes,
               libraries = ['mtx', 'modres']),
 
     Extension("pGroupCohomology.cochain",
               sources = [os.path.join("pGroupCohomology","cochain.pyx")],
-              include_dirs = sage_include_directories(),
+              include_dirs = includes,
               libraries = ['mtx', 'modres']),
 
     Extension("pGroupCohomology.cohomology",
               sources = [os.path.join("pGroupCohomology","cohomology.pyx")],
-              include_dirs = sage_include_directories(),
+              include_dirs = includes,
               libraries = ['mtx', 'modres']),
 
     Extension("pGroupCohomology.modular_cohomology",
               sources = [os.path.join("pGroupCohomology","modular_cohomology.pyx")],
-              include_dirs = sage_include_directories(),
+              include_dirs = includes,
               libraries = ['mtx', 'modres']),
 
     Extension("pGroupCohomology.dickson",
