diff -druN mpir-2.6.0.orig/configure.in mpir-2.6.0/configure.in
--- mpir-2.6.0.orig/configure.in	2012-11-08 23:10:45.000000000 +0100
+++ mpir-2.6.0/configure.in	2014-01-09 14:14:38.790341345 +0100
@@ -1180,38 +1176,38 @@
         esac
 
         case $host in
-		# NOTE x86_64w and x86_64 paths MUST be the same , for the fat structure
-          x86_64-w64-mingw*)
+          # NOTE x86_64w and x86_64 paths MUST be the same , for the fat structure
+          x86_64-w64-mingw*|x86_64-*-cygwin*)
             path_64="x86_64w" ;;
-          netburst-w64-mingw*)
+          netburst-w64-mingw*|netburst-*-cygwin*)
             path_64="x86_64w/netburst x86_64w" ;;
-          netburstlahf-w64-mingw*)
+          netburstlahf-w64-mingw*|netburstlahf-*-cygwin*)
             path_64="x86_64w/netburst x86_64w" ;;
-          k8-w64-mingw*)
+          k8-w64-mingw*|k8-*-cygwin*)
             path_64="x86_64w/k8/k8only x86_64w/k8 x86_64w" ;;
-          k10-w64-mingw*)
+          k10-w64-mingw*|k10-*-cygwin*)
             path_64="x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-          k102-w64-mingw*)
+          k102-w64-mingw*|k102-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  k103-w64-mingw*)
+          k103-w64-mingw*|k103-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  bulldozer-w64-mingw*)
+          bulldozer-w64-mingw*|bulldozer-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  bobcat-w64-mingw*)
+          bobcat-w64-mingw*|bobcat-*-cygwin*)
             path_64="x86_64w/bobcat x86_64w" ;;
-          core2-w64-mingw*) 
+          core2-w64-mingw*|core2-*-cygwin*)
             path_64="x86_64w/core2 x86_64w" ;;
-          penryn-w64-mingw*) 
+          penryn-w64-mingw*|penryn-*-cygwin*)
             path_64="x86_64w/core2/penryn x86_64w/core2 x86_64w" ;;
-          nehalem-w64-mingw*) 
+          nehalem-w64-mingw*|nehalem-*-cygwin*)
             path_64="x86_64w/nehalem x86_64w" ;;
-          westmere-w64-mingw*) 
+          westmere-w64-mingw*|westmere-*-cygwin*)
             path_64="x86_64w/nehalem/westmere x86_64w/nehalem x86_64w" ;;
-          sandybridge-w64-mingw*) 
+          sandybridge-w64-mingw*|sandybridge-*-cygwin*)
             path_64="x86_64w/sandybridge x86_64w" ;;
-          atom-w64-mingw*) 
+          atom-w64-mingw*|atom-*-cygwin*)
             path_64="x86_64w/atom x86_64w" ;;
-          nano-w64-mingw*) 
+          nano-w64-mingw*|nano-*-cygwin*)
             path_64="x86_64w/k8/k8only x86_64w/k8 x86_64w" ;;
 
           x86_64-*-*)
@@ -1253,14 +1249,14 @@
           gcc_64_cflags_cpu=""
           gcc_64_cflags_arch=""
           extra_functions_64="$extra_functions_64 fat fat_entry"
-	  case $host in
-		*-w64-mingw*)
-	          path_64="x86_64w x86_64w/fat"
-		  fat_path_64="x86_64w x86_64w/fat x86_64w/netburst x86_64w/k8 x86_64w/k8/k8only x86_64w/k8/k10 x86_64w/k8/k10/k102  x86_64w/core2 x86_64w/core2/penryn x86_64w/nehalem x86_64w/nehalem/westmere x86_64w/atom x86_64w/bobcat x86_64w/sandybridge" ;;
-		*-*-*)
-	          path_64="x86_64 x86_64/fat"
-        	  fat_path_64="x86_64 x86_64/fat x86_64/netburst x86_64/k8 x86_64/k8/k8only x86_64/k8/k10 x86_64/k8/k10/k102  x86_64/core2 x86_64/core2/penryn x86_64/nehalem x86_64/nehalem/westmere x86_64/atom x86_64/bobcat x86_64/sandybridge" ;;
-	  esac
+          case $host in
+            *-w64-mingw*|*-*-cygwin*)
+              path_64="x86_64w x86_64w/fat"
+              fat_path_64="x86_64w x86_64w/fat x86_64w/netburst x86_64w/k8 x86_64w/k8/k8only x86_64w/k8/k10 x86_64w/k8/k10/k102  x86_64w/core2 x86_64w/core2/penryn x86_64w/nehalem x86_64w/nehalem/westmere x86_64w/atom x86_64w/bobcat x86_64w/sandybridge" ;;
+            *-*-*)
+              path_64="x86_64 x86_64/fat"
+              fat_path_64="x86_64 x86_64/fat x86_64/netburst x86_64/k8 x86_64/k8/k8only x86_64/k8/k10 x86_64/k8/k10/k102  x86_64/core2 x86_64/core2/penryn x86_64/nehalem x86_64/nehalem/westmere x86_64/atom x86_64/bobcat x86_64/sandybridge" ;;
+          esac
         fi
         ;;
 
@@ -2612,25 +2608,27 @@
   fi
 done
 
-# for mingw64 rename all *.asm files to *.as files so we then will use yasm 
+# For win64 rename all *.asm files to *.as files so we then will use yasm 
 # cant move them yet as the mpn directory is not created until the end
 case $host in
-*-w64-mingw*)
-	old_gmp_srclinks=$gmp_srclinks
-	gmp_srclinks=
-	for jay in $old_gmp_srclinks; do
-		jayext=`echo "$jay" | awk -F . '{print $NF}'`
-		#jayext=`/bin/echo "${jay#*.}"`
-		#jayext=`echo "$jay" | sed -n 's/.*\.//'`
-		base=`echo "${jay%.*}"`
-		name=`basename $base`
-		if test "$jayext" = "asm" && test "$name" != "fat_entry"; then
-			gmp_srclinks="gmp_srclinks $base.as"
-		else
-			gmp_srclinks="$gmp_srclinks $jay"
-		fi
-	done
-	;;
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      old_gmp_srclinks=$gmp_srclinks
+      gmp_srclinks=
+      for jay in $old_gmp_srclinks; do
+        jayext=`echo "$jay" | awk -F . '{print $NF}'`
+        #jayext=`/bin/echo "${jay#*.}"`
+        #jayext=`echo "$jay" | sed -n 's/.*\.//'`
+        base=`echo "${jay%.*}"`
+        name=`basename $base`
+        if test "$jayext" = "asm" && test "$name" != "fat_entry"; then
+          gmp_srclinks="gmp_srclinks $base.as"
+        else
+          gmp_srclinks="$gmp_srclinks $jay"
+        fi
+      done
+    fi
+  ;;
 esac
 
 # All cycle counters are .asm files currently
@@ -2829,10 +2827,10 @@
           case $host in 
             *-*-darwin*)
               OBJECT_FORMAT="-f macho64" ;;
-	    *-w64-mingw*)
-		OBJECT_FORMAT="-f x64"  ;;
-	    *)
-	      OBJECT_FORMAT="-f elf64" ;;
+            *-w64-mingw*|*-*-cygwin*)
+              OBJECT_FORMAT="-f x64"  ;;
+            *)
+              OBJECT_FORMAT="-f elf64" ;;
           esac
           GMP_INCLUDE_MPN(x86_64/x86_64-defs.m4)
           ;;
@@ -2987,55 +2985,65 @@
 AC_CONFIG_SUBDIRS([yasm])
 
 case $host in
-  *-w64-mingw*)
-	if test $enable_fat = yes; then
-	YASM_MAC_INC=yasm_macwin.inc.fat
-	else
-	YASM_MAC_INC=yasm_macwin.inc.nofat
-	fi
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      if test $enable_fat = yes; then
+        YASM_MAC_INC=yasm_macwin.inc.fat
+      else
+        YASM_MAC_INC=yasm_macwin.inc.nofat
+      fi
+    else
+      if test $enable_fat = yes; then
+        YASM_MAC_INC=yasm_mac.inc.fat
+      else
+        YASM_MAC_INC=yasm_mac.inc.nofat
+      fi
+    fi
   ;;
   *-*-*)
-	if test $enable_fat = yes; then
-	YASM_MAC_INC=yasm_mac.inc.fat
-	else
-	YASM_MAC_INC=yasm_mac.inc.nofat
-	fi
+    if test $enable_fat = yes; then
+      YASM_MAC_INC=yasm_mac.inc.fat
+    else
+      YASM_MAC_INC=yasm_mac.inc.nofat
+    fi
   ;;
 esac
 AC_CONFIG_LINKS(yasm_mac.inc:$YASM_MAC_INC)
 
 AC_OUTPUT(Makefile mpf/Makefile mpn/Makefile fft/Makefile mpq/Makefile mpz/Makefile printf/Makefile scanf/Makefile cxx/Makefile tests/Makefile tests/devel/Makefile tests/mpf/Makefile tests/mpn/Makefile tests/fft/Makefile tests/mpq/Makefile tests/mpz/Makefile tests/rand/Makefile tests/misc/Makefile tests/cxx/Makefile doc/Makefile tune/Makefile mpir.h:gmp-h.in)
 if test $enable_gmpcompat = yes; then
 	AC_OUTPUT(gmp.h:gmp-h.in)
 	if test $enable_cxx = yes ; then
 		AC_OUTPUT(gmpxx.h:mpirxx.h)
 	fi
 fi
 # now we have created and populated the mpn directory we can rename the files
 # and post process the "m4 macros" into "yasm macros"
 case $host in
-*-w64-mingw*)
-	for jay in $old_gmp_srclinks; do
-		jayext=`echo "$jay" | awk -F . '{print $NF}'`
-		#jayext=`/bin/echo "${jay#*.}"`
-		#jayext=`echo "$jay" | sed -n 's/.*\.//'`
-		base=`echo "${jay%.*}"`
-		name=`basename $base`
-		if test "$jayext" = "asm" && test "$name" != "fat_entry" ; then
-			got=`grep -ce "d""nl  This line is to help MINGW64" $jay`
-			if test "$got" = "1" ; then
-				cat $jay | sed s/"include(\(.*\))"/"%include \"\1\""/ | sed s/"d""efine(\(.*\),\(.*\))"/"%d""efine \1 \2"/ | sed s/"d""efine(\(.*\))"/"%d""efine \1"/ | sed s/"d""nl\(.*\)"/"; \1"/ | sed s/ifdef.*// | sed s/"\`.*"// > $base.as
-				rm -f $jay
-			else
-				mv $jay $base.as
-			fi
-		fi
-	done
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      for jay in $old_gmp_srclinks; do
+        jayext=`echo "$jay" | awk -F . '{print $NF}'`
+        #jayext=`/bin/echo "${jay#*.}"`
+        #jayext=`echo "$jay" | sed -n 's/.*\.//'`
+        base=`echo "${jay%.*}"`
+        name=`basename $base`
+        if test "$jayext" = "asm" && test "$name" != "fat_entry" ; then
+          got=`grep -ce "d""nl  This line is to help MINGW64" $jay`
+          if test "$got" = "1" ; then
+            cat $jay | sed s/"include(\(.*\))"/"%include \"\1\""/ | sed s/"d""efine(\(.*\),\(.*\))"/"%d""efine \1 \2"/ | sed s/"d""efine(\(.*\))"/"%d""efine \1"/ | sed s/"d""nl\(.*\)"/"; \1"/ | sed s/ifdef.*// | sed s/"\`.*"// > $base.as
+            rm -f $jay
+          else
+            mv $jay $base.as
+          fi
+        fi
+      done
 # the file x86_64_divrem_euclidean_qr_1.as should detect the defined macro EXCLUDE_PREINV , but for some reason it doesnt, so we hack it
-	if test $enable_fat = yes; then
-		echo "%d""efine EXCLUDE_PREINV 1" > mpn/x86_64_divrem_euclidean_qr_1.tmp
-		cat mpn/x86_64_divrem_euclidean_qr_1.as >> mpn/x86_64_divrem_euclidean_qr_1.tmp
-		mv mpn/x86_64_divrem_euclidean_qr_1.tmp mpn/x86_64_divrem_euclidean_qr_1.as
-	fi
-;;
+      if test $enable_fat = yes; then
+        echo "%d""efine EXCLUDE_PREINV 1" > mpn/x86_64_divrem_euclidean_qr_1.tmp
+        cat mpn/x86_64_divrem_euclidean_qr_1.as >> mpn/x86_64_divrem_euclidean_qr_1.tmp
+        mv mpn/x86_64_divrem_euclidean_qr_1.tmp mpn/x86_64_divrem_euclidean_qr_1.as
+      fi
+    fi
+  ;;
 esac
diff -druN mpir-2.6.0.orig/configure mpir-2.6.0/configure
--- mpir-2.6.0.orig/configure	2012-11-08 23:13:27.000000000 +0100
+++ mpir-2.6.0/configure	2014-01-09 14:14:38.438343758 +0100
@@ -4747,38 +4874,38 @@
         esac
 
         case $host in
-		# NOTE x86_64w and x86_64 paths MUST be the same , for the fat structure
-          x86_64-w64-mingw*)
+          # NOTE x86_64w and x86_64 paths MUST be the same , for the fat structure
+          x86_64-w64-mingw*|x86_64-*-cygwin*)
             path_64="x86_64w" ;;
-          netburst-w64-mingw*)
+          netburst-w64-mingw*|netburst-*-cygwin*)
             path_64="x86_64w/netburst x86_64w" ;;
-          netburstlahf-w64-mingw*)
+          netburstlahf-w64-mingw*|netburstlahf-*-cygwin*)
             path_64="x86_64w/netburst x86_64w" ;;
-          k8-w64-mingw*)
+          k8-w64-mingw*|k8-*-cygwin*)
             path_64="x86_64w/k8/k8only x86_64w/k8 x86_64w" ;;
-          k10-w64-mingw*)
+          k10-w64-mingw*|k10-*-cygwin*)
             path_64="x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-          k102-w64-mingw*)
+          k102-w64-mingw*|k102-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  k103-w64-mingw*)
+          k103-w64-mingw*|k103-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  bulldozer-w64-mingw*)
+          bulldozer-w64-mingw*|bulldozer-*-cygwin*)
             path_64="x86_64w/k8/k10/k102 x86_64w/k8/k10 x86_64w/k8 x86_64w" ;;
-	  bobcat-w64-mingw*)
+          bobcat-w64-mingw*|bobcat-*-cygwin*)
             path_64="x86_64w/bobcat x86_64w" ;;
-          core2-w64-mingw*)
+          core2-w64-mingw*|core2-*-cygwin*)
             path_64="x86_64w/core2 x86_64w" ;;
-          penryn-w64-mingw*)
+          penryn-w64-mingw*|penryn-*-cygwin*)
             path_64="x86_64w/core2/penryn x86_64w/core2 x86_64w" ;;
-          nehalem-w64-mingw*)
+          nehalem-w64-mingw*|nehalem-*-cygwin*)
             path_64="x86_64w/nehalem x86_64w" ;;
-          westmere-w64-mingw*)
+          westmere-w64-mingw*|westmere-*-cygwin*)
             path_64="x86_64w/nehalem/westmere x86_64w/nehalem x86_64w" ;;
-          sandybridge-w64-mingw*)
+          sandybridge-w64-mingw*|sandybridge-*-cygwin*)
             path_64="x86_64w/sandybridge x86_64w" ;;
-          atom-w64-mingw*)
+          atom-w64-mingw*|atom-*-cygwin*)
             path_64="x86_64w/atom x86_64w" ;;
-          nano-w64-mingw*)
+          nano-w64-mingw*|nano-*-cygwin*)
             path_64="x86_64w/k8/k8only x86_64w/k8 x86_64w" ;;
 
           x86_64-*-*)
@@ -4820,14 +4947,14 @@
           gcc_64_cflags_cpu=""
           gcc_64_cflags_arch=""
           extra_functions_64="$extra_functions_64 fat fat_entry"
-	  case $host in
-		*-w64-mingw*)
-	          path_64="x86_64w x86_64w/fat"
-		  fat_path_64="x86_64w x86_64w/fat x86_64w/netburst x86_64w/k8 x86_64w/k8/k8only x86_64w/k8/k10 x86_64w/k8/k10/k102  x86_64w/core2 x86_64w/core2/penryn x86_64w/nehalem x86_64w/nehalem/westmere x86_64w/atom x86_64w/bobcat x86_64w/sandybridge" ;;
-		*-*-*)
-	          path_64="x86_64 x86_64/fat"
-        	  fat_path_64="x86_64 x86_64/fat x86_64/netburst x86_64/k8 x86_64/k8/k8only x86_64/k8/k10 x86_64/k8/k10/k102  x86_64/core2 x86_64/core2/penryn x86_64/nehalem x86_64/nehalem/westmere x86_64/atom x86_64/bobcat x86_64/sandybridge" ;;
-	  esac
+          case $host in
+            *-w64-mingw*|*-*-cygwin*)
+              path_64="x86_64w x86_64w/fat"
+              fat_path_64="x86_64w x86_64w/fat x86_64w/netburst x86_64w/k8 x86_64w/k8/k8only x86_64w/k8/k10 x86_64w/k8/k10/k102  x86_64w/core2 x86_64w/core2/penryn x86_64w/nehalem x86_64w/nehalem/westmere x86_64w/atom x86_64w/bobcat x86_64w/sandybridge" ;;
+            *-*-*)
+              path_64="x86_64 x86_64/fat"
+              fat_path_64="x86_64 x86_64/fat x86_64/netburst x86_64/k8 x86_64/k8/k8only x86_64/k8/k10 x86_64/k8/k10/k102  x86_64/core2 x86_64/core2/penryn x86_64/nehalem x86_64/nehalem/westmere x86_64/atom x86_64/bobcat x86_64/sandybridge" ;;
+          esac
         fi
         ;;
 
@@ -26783,25 +27092,27 @@
   fi
 done
 
-# for mingw64 rename all *.asm files to *.as files so we then will use yasm
+# For win64 rename all *.asm files to *.as files so we then will use yasm
 # cant move them yet as the mpn directory is not created until the end
 case $host in
-*-w64-mingw*)
-	old_gmp_srclinks=$gmp_srclinks
-	gmp_srclinks=
-	for jay in $old_gmp_srclinks; do
-		jayext=`echo "$jay" | awk -F . '{print $NF}'`
-		#jayext=`/bin/echo "${jay#*.}"`
-		#jayext=`echo "$jay" | sed -n 's/.*\.//'`
-		base=`echo "${jay%.*}"`
-		name=`basename $base`
-		if test "$jayext" = "asm" && test "$name" != "fat_entry"; then
-			gmp_srclinks="gmp_srclinks $base.as"
-		else
-			gmp_srclinks="$gmp_srclinks $jay"
-		fi
-	done
-	;;
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      old_gmp_srclinks=$gmp_srclinks
+      gmp_srclinks=
+      for jay in $old_gmp_srclinks; do
+        jayext=`echo "$jay" | awk -F . '{print $NF}'`
+        #jayext=`/bin/echo "${jay#*.}"`
+        #jayext=`echo "$jay" | sed -n 's/.*\.//'`
+        base=`echo "${jay%.*}"`
+        name=`basename $base`
+        if test "$jayext" = "asm" && test "$name" != "fat_entry"; then
+          gmp_srclinks="gmp_srclinks $base.as"
+        else
+          gmp_srclinks="$gmp_srclinks $jay"
+        fi
+      done
+    fi
+  ;;
 esac
 
 # All cycle counters are .asm files currently
@@ -28066,10 +28377,10 @@
           case $host in
             *-*-darwin*)
               OBJECT_FORMAT="-f macho64" ;;
-	    *-w64-mingw*)
-		OBJECT_FORMAT="-f x64"  ;;
-	    *)
-	      OBJECT_FORMAT="-f elf64" ;;
+            *-w64-mingw*|*-*-cygwin*)
+              OBJECT_FORMAT="-f x64"  ;;
+            *)
+              OBJECT_FORMAT="-f elf64" ;;
           esac
 
 echo "include_mpn(\`x86_64/x86_64-defs.m4')" >> $gmp_tmpconfigm4i
@@ -28438,19 +28749,27 @@
 
 
 case $host in
-  *-w64-mingw*)
-	if test $enable_fat = yes; then
-	YASM_MAC_INC=yasm_macwin.inc.fat
-	else
-	YASM_MAC_INC=yasm_macwin.inc.nofat
-	fi
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      if test $enable_fat = yes; then
+        YASM_MAC_INC=yasm_macwin.inc.fat
+      else
+        YASM_MAC_INC=yasm_macwin.inc.nofat
+      fi
+    else
+      if test $enable_fat = yes; then
+        YASM_MAC_INC=yasm_mac.inc.fat
+      else
+        YASM_MAC_INC=yasm_mac.inc.nofat
+      fi
+    fi
   ;;
   *-*-*)
-	if test $enable_fat = yes; then
-	YASM_MAC_INC=yasm_mac.inc.fat
-	else
-	YASM_MAC_INC=yasm_mac.inc.nofat
-	fi
+    if test $enable_fat = yes; then
+      YASM_MAC_INC=yasm_mac.inc.fat
+    else
+      YASM_MAC_INC=yasm_mac.inc.nofat
+    fi
   ;;
 esac
 ac_config_links="$ac_config_links yasm_mac.inc:$YASM_MAC_INC"
@@ -37530,33 +37795,35 @@
 $as_echo "$as_me: WARNING: unrecognized options: $ac_unrecognized_opts" >&2;}
 fi
 
-	fi
+  fi
 fi
 # now we have created and populated the mpn directory we can rename the files
 # and post process the "m4 macros" into "yasm macros"
 case $host in
-*-w64-mingw*)
-	for jay in $old_gmp_srclinks; do
-		jayext=`echo "$jay" | awk -F . '{print $NF}'`
-		#jayext=`/bin/echo "${jay#*.}"`
-		#jayext=`echo "$jay" | sed -n 's/.*\.//'`
-		base=`echo "${jay%.*}"`
-		name=`basename $base`
-		if test "$jayext" = "asm" && test "$name" != "fat_entry" ; then
-			got=`grep -ce "d""nl  This line is to help MINGW64" $jay`
-			if test "$got" = "1" ; then
-				cat $jay | sed s/"include(\(.*\))"/"%include \"\1\""/ | sed s/"d""efine(\(.*\),\(.*\))"/"%d""efine \1 \2"/ | sed s/"d""efine(\(.*\))"/"%d""efine \1"/ | sed s/"d""nl\(.*\)"/"; \1"/ | sed s/ifdef.*// | sed s/"\`.*"// > $base.as
-				rm -f $jay
-			else
-				mv $jay $base.as
-			fi
-		fi
-	done
+  *-w64-mingw*|*-*-cygwin*)
+    if test "$ABI" = "64"; then
+      for jay in $old_gmp_srclinks; do
+        jayext=`echo "$jay" | awk -F . '{print $NF}'`
+        #jayext=`/bin/echo "${jay#*.}"`
+        #jayext=`echo "$jay" | sed -n 's/.*\.//'`
+        base=`echo "${jay%.*}"`
+        name=`basename $base`
+        if test "$jayext" = "asm" && test "$name" != "fat_entry" ; then
+          got=`grep -ce "d""nl  This line is to help MINGW64" $jay`
+          if test "$got" = "1" ; then
+            cat $jay | sed s/"include(\(.*\))"/"%include \"\1\""/ | sed s/"d""efine(\(.*\),\(.*\))"/"%d""efine \1 \2"/ | sed s/"d""efine(\(.*\))"/"%d""efine \1"/ | sed s/"d""nl\(.*\)"/"; \1"/ | sed s/ifdef.*// | sed s/"\`.*"// > $base.as
+            rm -f $jay
+          else
+            mv $jay $base.as
+          fi
+        fi
+      done
 # the file x86_64_divrem_euclidean_qr_1.as should detect the defined macro EXCLUDE_PREINV , but for some reason it doesnt, so we hack it
-	if test $enable_fat = yes; then
-		echo "%d""efine EXCLUDE_PREINV 1" > mpn/x86_64_divrem_euclidean_qr_1.tmp
-		cat mpn/x86_64_divrem_euclidean_qr_1.as >> mpn/x86_64_divrem_euclidean_qr_1.tmp
-		mv mpn/x86_64_divrem_euclidean_qr_1.tmp mpn/x86_64_divrem_euclidean_qr_1.as
-	fi
-;;
+      if test $enable_fat = yes; then
+        echo "%d""efine EXCLUDE_PREINV 1" > mpn/x86_64_divrem_euclidean_qr_1.tmp
+        cat mpn/x86_64_divrem_euclidean_qr_1.as >> mpn/x86_64_divrem_euclidean_qr_1.tmp
+        mv mpn/x86_64_divrem_euclidean_qr_1.tmp mpn/x86_64_divrem_euclidean_qr_1.as
+      fi
+    fi
+  ;;
 esac
