diff --git a/field/gf2.h b/field/gf2.h
index 97c039b..2a1a7d6 100644
--- a/linbox/field/gf2.h
+++ b/linbox/field/gf2.h
@@ -41,7 +41,7 @@
 #include "linbox/field/field-traits.h"
 // #include "linbox/vector/vector-domain.h"
 
-#if !defined(__PATHCC__) && !(defined(__APPLE__) && defined(__clang__))
+#if !defined(__PATHCC__) && !((defined(__APPLE__) || defined(__FreeBSD__)) && defined(__clang__))
 #define stdBitReference std::_Bit_reference
 #else
 #define stdBitReference std::vector<bool>::reference
@@ -1004,7 +1004,7 @@ namespace LinBox
 #include "linbox/randiter/gf2.h"
 
 #if __cplusplus >= 201103L
-#if defined( __APPLE__) && defined(__clang__)
+#if (defined(__APPLE__) || defined(__FreeBSD__)) && defined(__clang__)
 #include <__bit_reference>
 #else
 #include <bits/stl_bvector.h>
